### PR TITLE
Update documentations on snapshots

### PR DESCRIPTION
**Pull request by cortinico**
I'm updating the documentation around using SNAPSHOTS as it was lacking.

Moreover I'm updating the SNAPSHOT name to be `master-SNAPSHOT`. If we don't like it we could use the upcoming version (but that forces us to keep two versions number updated).

### CODE REVIEW

**Review by BraisGabin — APPROVED** [review]
👏🏻👏🏻👏🏻👏🏻 Nice one! And I like the `master-SNAPSHOT` name.

File: docs/pages/snapshots.md [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    repositories {
+        // Your other repos here.
+        maven {
+            url = uri("https://oss.sonatype.org/content/repositories/snapshots/")
+        }
+    }
+}
+```
+
+Please note that the extra `resolutionStrategy{}` block is needed as we don't publish a Gradle Plugin marker for our snapshots.
```
Comment (chao2zhang) [inline docs/pages/snapshots.md:82]: I came to read https://docs.gradle.org/current/userguide/plugins.html#sec:plugin_markers. I am wondering why we don't publish a Gradle Plugin marker in our build?

**Review by chao2zhang — APPROVED** [review]
I have one question if you may know the answer to.

File: docs/pages/snapshots.md (hunk 1 above)
Comment (cortinico) [reply docs/pages/snapshots.md:82]: The reason is here: https://github.com/detekt/detekt/blob/1164c34f1ce1bd4f8f4a6df57d7599798f1ff06d/detekt-gradle-plugin/build.gradle.kts#L58-L60

I'm unsure of what was the exact issue with `isAutomatedPublishing` and the invalid signature, but we can definitely look into it.

File: docs/pages/snapshots.md [hunk 2]
```diff
@@ -0,0 +1,82 @@
+---
+title: "Using Snapshots"
+keywords: snapshot releases 
+sidebar: 
+permalink: snapshots.html
+summary: This page explains you how you can setup snapshots for your Detekt build to test the latest unreleased features. 
+---
+
+This page explains you how you can use our **latest snapshots**, to test upcoming unreleased features.
```
Comment (schalkms) [inline docs/pages/snapshots.md:9]:
```suggestion
This page explains how you can use our **latest snapshots** to test upcoming unreleased features.
```

File: docs/pages/snapshots.md [hunk 3]
```diff
@@ -0,0 +1,82 @@
+---
+title: "Using Snapshots"
+keywords: snapshot releases 
+sidebar: 
+permalink: snapshots.html
+summary: This page explains you how you can setup snapshots for your Detekt build to test the latest unreleased features.
```
Comment (schalkms) [inline docs/pages/snapshots.md:6]:
```suggestion
summary: This page explains how you can setup snapshots for your Detekt build to test the latest unreleased features. 
```

**Review by schalkms — APPROVED** [review]

**Pull request merged**